tasks/libs/common/git.py

```diff
@@ -148,10 +148,32 @@ def get_default_branch(major: int | None = None):
     return '6.53.x' if major is None and is_agent6(ctx) or major == 6 else 'main'
 
 
-def get_common_ancestor(ctx, branch, base=None) -> str:
-    base = base or f"origin/{get_default_branch()}"
+def get_common_ancestor(ctx, branch, base=None, try_fetch=True) -> str:
+    """
+    Get the common ancestor between two branches.
+
+    Args:
+        ctx: The invoke context.
+        branch: The branch to get the common ancestor with.
+        base: The base branch to get the common ancestor with.
+        try_fetch: Try to fetch the base branch if it's not found (to avoid S3 caching issues).
+
+    Returns:
+        The common ancestor between two branches.
+    """
+
+    base = (base or get_default_branch()).removeprefix("origin/")
```

Comment:
Should it default to the base branch instead of the default branch ? Main difference being on release branches